### DISCUSSION TITLE
Font rendering priority

### DISCUSSION THREAD

**Discussion by Jacherr** [libvips/libvips#4023] [Q&A]
Hi! I'm attempting to render some text using an emoji font loaded from `/usr/share/fonts`. I do this by passing `Twemoji Color Emoji` to the `font` parameter on `vips_text`. However, despite specifying this font and having cached it with `fc-cache -fv`, the renderer still falls back to using the system's emoji font (in this case Google noto). FWIW, the font also shows up in `fc-list`.

If I change fontconfig to manually give Twemoji a higher priority it does then work, but that then sets my system emoji font to twemoji, which isn't what I'm after. Is there some step I'm missing? I also tried the suggestion [here](https://github.com/libvips/libvips/issues/3023#issuecomment-1236318876), but this didn't seem to help much either.

**Comment by jcupitt** [comment]
Hello @Jacherr,

I'd think just specifying the font name should do the trick. Have you tried with `FC_DEBUG`? It prints a HUGE trace of the steps it took in font resolution, eg.:

```
$ FC_DEBUG=1 vips text x.png "hello world" --font mono | wc
    387    2559   25351
```

... 387 lines of logging explaining how it settled on `DejaVu Sans Mono`.

**Comment by Jacherr** [comment]
I have had a go but to be fair this is very hard to read 😅, it just seems to be sorting the Noto emoji before the Twemoji. It's not a critical issue because this only happens on my local machine where there are multiple emoji fonts installed! I'll see if I can figure out what this output is trying tot tell me..
